numpy/core/src/umath/loops_unary_fp_le.dispatch.c.src

```diff
@@ -0,0 +1,556 @@
+#include "fast_loop_macros.h"
+
+
+/*******************************************************************************
+ ** extra SIMD intrinsics
+ ******************************************************************************/
+
+#if NPY_SIMD
+/**begin repeat
+ * #TYPE = FLOAT, DOUBLE#
+ * #sfx  = f32, f64#
+ * #VCHK = NPY_SIMD_F32, NPY_SIMD_F64#
+ * #FDMAX = FLT_MAX, DBL_MAX#
+ * #fd = f, #
+ * #ssfx = 32, 64#
+ */
+#if @VCHK@
+
+static NPY_INLINE npyv_u@ssfx@
+npyv_isnan_@sfx@(npyv_@sfx@ v)
+{
+    // (v != v) >> (size - 1)
+#if defined(NPY_HAVE_SSE2) || defined (NPY_HAVE_SSE41)
+    // sse npyv_cmpneq_@sfx@ define includes a cast already
+    npyv_u@ssfx@ r = npyv_cmpneq_@sfx@(v, v);
+#else
+    npyv_u@ssfx@ r = npyv_cvt_u@ssfx@_b@ssfx@(npyv_cmpneq_@sfx@(v, v));
+#endif
+#if defined(NPY__CPU_TARGET_VX) || defined(NPY__CPU_TARGET_VXE)
+    // don't do the shift on s390x
+    return r;
+#else
+    return npyv_shri_u@ssfx@(r, (sizeof(npyv_lanetype_@sfx@)*8)-1);
+#endif
+}
+
+static NPY_INLINE npyv_u@ssfx@
+npyv_isinf_@sfx@(npyv_@sfx@ v)
+{
+    // (abs(v) > fltmax) >> (size - 1)
+    const npyv_@sfx@ fltmax = npyv_setall_@sfx@(@FDMAX@);
+#if defined(NPY_HAVE_NEON)
+    npyv_u@ssfx@ r = vcagtq_@sfx@(v, fltmax);
+#else
+    // fabs via masking of sign bit
+    const npyv_@sfx@ signmask = npyv_setall_@sfx@(-0.@fd@);
+    npyv_u8      r_u8 = npyv_andc_u8(npyv_reinterpret_u8_@sfx@(v), npyv_reinterpret_u8_@sfx@(signmask));
+ #if defined(NPY_HAVE_SSE2) || defined (NPY_HAVE_SSE41)
+    // return cast already done in npyv_cmpgt_@sfx@
+    npyv_u@ssfx@ r    = npyv_cmpgt_@sfx@(npyv_reinterpret_@sfx@_u8(r_u8), fltmax);
+ #else
+    npyv_u@ssfx@ r    = npyv_reinterpret_u@ssfx@_@sfx@(npyv_cmpgt_@sfx@(npyv_reinterpret_@sfx@_u8(r_u8), fltmax));
+ #endif
+#endif
+#if defined(NPY__CPU_TARGET_VX) || defined(NPY__CPU_TARGET_VXE)
+    // don't do the shift on s390x
+    return r;
+#else
+    return npyv_shri_u@ssfx@(r, (sizeof(npyv_lanetype_@sfx@)*8)-1);
+#endif
+}
+
+static NPY_INLINE npyv_u@ssfx@
+npyv_isfinite_@sfx@(npyv_@sfx@ v)
+{
+    // ((v & signmask) <= fltmax) >> (size-1)
+    const npyv_@sfx@ fltmax = npyv_setall_@sfx@(@FDMAX@);
+#if defined(NPY_HAVE_NEON)
+    npyv_u@ssfx@ r = vcaleq_@sfx@(v, fltmax);
+#else
+    // fabs via masking of sign bit
+    const npyv_@sfx@ signmask = npyv_setall_@sfx@(-0.@fd@);
+    npyv_u8      r_u8 = npyv_andc_u8(npyv_reinterpret_u8_@sfx@(v), npyv_reinterpret_u8_@sfx@(signmask));
+ #if defined(NPY_HAVE_SSE2) || defined (NPY_HAVE_SSE41)
+    // return cast already done in npyv_cmpgt_@sfx@
+    npyv_u@ssfx@ r    = npyv_cmple_@sfx@(npyv_reinterpret_@sfx@_u8(r_u8), fltmax);
+ #else
+    npyv_u@ssfx@ r    = npyv_reinterpret_u@ssfx@_@sfx@(npyv_cmple_@sfx@(npyv_reinterpret_@sfx@_u8(r_u8), fltmax));
+ #endif
+#endif
+#if defined(NPY__CPU_TARGET_VX) || defined(NPY__CPU_TARGET_VXE)
+    // don't do the shift on s390x
+    return r;
+#else
+    return npyv_shri_u@ssfx@(r, (sizeof(npyv_lanetype_@sfx@)*8)-1);
+#endif
+}
```

Comment:
```suggestion
#if NPY_SIMD_F32
NPY_FINLINE npyv_b32
npyv_notfinite_f32(npyv_f32 a)
{
    // cast out the sign and check if all exponent bits are set
    // no matter the mentissa is.
    const npyv_u32 exp_mask = npyv_setall_u32(0xff000000);
    npyv_u32 bits = npyv_reinterpret_u32_f32(a);
    bits = npyv_shli_u32(a, 1);
    bits = npyv_and_u32(bits, exp_mask);
    return npyv_cmpeq_u32(bits, exp_mask);
}
#endif // NPY_SIMD_F32
```
same idea as `isinf` but discard the mantissa bits and `notfinite` rather than `isfinite` similar to `notnan`.